### PR TITLE
[ONNX] Bump ONNX Runtime version to 1.12.1 in CI

### PR DESCRIPTION
**Pull request by BowenBao**
Bump ONNX Runtime version to 1.12 in CI to support testing ONNX version 17 ops.

### CODE REVIEW

**Comment by BowenBao** [comment]
cc @abock for approval review.

**Review by abock — APPROVED** [review]
Amazing. Best patch ever.

**Review by titaiwangms — APPROVED** [review]
🍻

File: .jenkins/caffe2/test.sh [hunk 1]
```diff
@@ -173,7 +173,7 @@ fi
 ##############
 if [[ "$BUILD_ENVIRONMENT" == *onnx* ]]; then
   pip install -q --user --no-use-pep517 "git+https://github.com/pytorch/vision.git@$(cat .github/ci_commit_pins/vision.txt)"
-  pip install -q --user ninja flatbuffers==2.0 numpy==1.21.5 onnxruntime==1.11.0
+  pip install -q --user ninja flatbuffers==2.0 numpy==1.21.5 onnxruntime==1.12.0
```
Comment (BowenBao) [inline .jenkins/caffe2/test.sh:176]:
```suggestion
  pip install -q --user ninja flatbuffers==2.0 numpy==1.21.5 onnxruntime==1.12.1
```

**Pull request closed without merging**